src/loader.cpp

```diff
@@ -0,0 +1,56 @@
+#include "loader.hpp"
+
+InstructionLoader::InstructionLoader(const std::string &trace_filename) {
+    LoadTraceFile(trace_filename);
+}
+
+InstructionLoader::~InstructionLoader() {
+    if (in_file == nullptr) {
+        in_file->close();
+    }
+}
+
+void InstructionLoader::LoadTraceFile(const std::string &filename) {
+    in_file = std::make_unique<std::ifstream>(filename.c_str(), std::ios::in);
+    if (in_file->fail()) {
+        std::cerr << "Open trace file error" << std::endl;
+        exit(-1);
+    }
+}
+
+inst_t InstructionLoader::GetNextInst() {
+    int LENGTH_OF_INST_LINE = 13;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'LENGTH_OF_INST_LINE' can be reduced.](https://app.codacy.com/app/jerry0715no14/Cache_sim/pullRequest?prid=4720103)